crates/turborepo-cache/src/lib.rs

```diff
@@ -82,8 +80,46 @@ pub enum CacheSource {
     Remote,
 }
 
-#[derive(Debug, Clone, PartialEq, Copy)]
-pub struct CacheResponse {
+// This is generic because exists returns just the metadata
+// while fetch returns the metadata and the files
+#[derive(Clone, Debug, PartialEq)]
+pub enum CacheResult<T> {
+    Miss,
+    Hit(T),
+}
```

Comment:
I think we should just get rid of this bespoke `Option` implementation unless there's some benefit we get from this that I'm not understanding.